code/modules/mob/living/simple_animal/bot/bot_construction.dm

```diff
@@ -387,8 +391,11 @@
 	if(is_pen(I))
 		var/t = rename_interactive(user, I, prompt = "Enter new robot name")
 		if(!isnull(t))
```

Comment:
I'd recommend replacing the nullcheck with the length check, and moving everything back an indent. `length(null)` == 0.

You should be able to do that for all of these.